internal/backend/remote-state/s3/backend.go

```diff
@@ -674,6 +717,27 @@ func configureAssumeRole(obj cty.Value) *awsbase.AssumeRole {
 	return &assumeRole
 }
 
+func configureAssumeRoleWithWebIdentity(obj cty.Value) *awsbase.AssumeRoleWithWebIdentity {
+	cfg := &awsbase.AssumeRoleWithWebIdentity{
+		RoleARN:              stringAttrDefaultEnvVar(obj, "role_arn", "AWS_ROLE_ARN"),
+		Policy:               stringAttr(obj, "policy"),
+		PolicyARNs:           stringSliceAttr(obj, "policy_arns"),
+		SessionName:          stringAttrDefaultEnvVar(obj, "session_name", "AWS_ROLE_SESSION_NAME"),
+		WebIdentityToken:     stringAttr(obj, "web_identity_token"),
```

Comment:
Why wouldn't we want to be able to pass this via an envvar? We do it for the token file, but not for the token itself, which looks inconsistent and potentially cumbersome to use?

Reply:
Afaik the idea behind the new APIs is to move away from environment-variable-based configuration, and instead use e.g. the aws config file present on the machine.

Reply:
Well but all the other fields have corresponding envvars.

Reply:
Fair.

This actually seems to correspond to the AWS CLI though.

The AWS CLI also doesn't expose that one as an env var, but does the others: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html

Reply:
So it might be just consistency with that.

Reply:
I used the env variables from the [guide](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-role.html#cli-configure-role-oidc) but I can't think of a reason not to add it.

Reply:
I think we can add it, then.